modules/tensor_mechanics/doc/content/documentation/systems/Materials/ComputeMultipleInelasticStress.md

```diff
@@ -1,14 +1,242 @@
+               in the case when only a single inelastic material model is specified.
+
+The algorithm, shown in [fig:single_material], used for a single inelastic material
+model is an optimized version of the multiple materials algorithm.
+With no need to iterate over multiple inelastic models, both the inner and outer
+iterations from [fig:multiple_materials] are removed from the alorithm in [fig:single_material].
+
+The initial elastic strain increment guess is assumed to be the initial mechanical
+strian increment, and the trial stress for the single inelastic model is calculated
+from that elastic strain increment as in [eqn:trial_stress].
+These stress and strain values are passed directly to the inelastic material model.
+
+The material model computes the admissble stress and strain states, as indicated
+by the blue element in [fig:single_material]. An optional consistent tangent
+operator matrix is also returned by the inelastic material model.
+The consistent tangent operator is then used to find the Jacobian multiplier with
+one of the three methods given in [eqn:combined_jacobian_mult], where the value
+of $i$ is limited to 1 in this case.
+
+The maximum size of the allowable time step is then optionally calculated by the
+inelastic material model, as described in the section below on the Material Time
+Step Limiter. At the conclusion of the algorithm, the value of the elastic and
+inelastic strain states are updated as shown in [fig:single_material].
+
+### Cycle Through One Inelastic Model per Time Step
+
+`ComputeMultipleInelasticStress` also includes an option to run a series of inelastic
+models in a rotating fashion such that only a single inelastic model is run on a
+timestep. This option uses the same algorithm as in [fig:single_material] to determine
+the strains and stress value based on the rotated single inelastic model.
+A separate method is then employed to propagate the strain and stress values to
+the other inelastic material models for storage as old material property values.
+
+## Helper Methods
+
+The `ComputeMultipleInelasticStress` material relies on two helper methods to
+aid the simulation in converging. The first helper method, the consistent tangent
+operator, is an optional feature which is implemented for only certain inelastic
+stress material models, and the material time step limiter is implemented in the
+models which use the [Radial Return Stress Update](/RadialReturnStressUpdate.md)
+algorithm.
+
+### Consistent Tangent Operator
+
+The consistent tangent operator is used to improve the convergence of mechanics
+problems (see a reference such as [cite:simo1985cto] for an introduction to
+consistent tangent operators). The value of the consistent tangent operator
+is computed in each inelastic material model. By default the consistent tangent
+operator is set equal to the elasticity tensor (the option Elastic in [eqn:combined_jacobian_mult]).
```

Comment:
This gets a big tricky, but by default, the option to calculate the full tangent operator is defaulted to true `tangent_operator=nonlinear`. However, each individual model must specify which type of tangent it calculates (full, partial, or none). Full models only work with none and/or none models, and partial models only work with partial and/or none models. PowerLawCreepStressUpdate has an example of a partial tangent operator implementation.

Reply:
I'll add a citation once we have a good one down the road.

Reply:
Could you clarify this for me please @tophmatthews, particularly which models the full tangent operator calculation works with?